application/deed/views.py

```diff
@@ -106,3 +108,51 @@ def delete_borrower(borrower_id):
         abort(status.HTTP_404_NOT_FOUND)
     else:
         return jsonify({"id": borrower_id}), status.HTTP_200_OK
+
+
+@deed_bp.route('/<deed_reference>/sign', methods=['POST'])
+def sign_deed(deed_reference):
+    request_json = request.get_json()
+    borrower_token = request_json['borrower_token']
+    result = Deed.query.filter_by(token=str(deed_reference)).first()
+
```

Comment:
This should be protected with the header id like the main get endpoint. Calling the get method on the model will do the header parsing for you.